config/nrfconnect/chip-module/Kconfig.features

```diff
@@ -230,4 +230,39 @@ config CHIP_WIFI_CONNECTION_RECOVERY_JITTER
 	  a random jitter interval is added to it to avoid periodicity. The random jitter is selected
 	  within range [-JITTER; +JITTER].
 
+choice CHIP_LAST_FABRIC_REMOVED_ACTION
+	prompt "An action to perform after removing the last fabric"
+	default CHIP_LAST_FABRIC_REMOVED_ERASE_AND_REBOOT
+
+	config CHIP_LAST_FABRIC_REMOVED_NONE
+	bool "After removing the last fabric do not perform any action"
+	help
+	  After removing the last fabric the device will not perform factory reset
+	  or reboot. The current state will be left as it is and the BLE advertising
+	  will not start automatically.
```

Comment:
But per spec, network credentials that were added via Matter should be removed, right?

Reply:
Right, however, there might be instances where the Thread device was commissioned to the Thread network out-of-band in the first place.

Btw. My understanding (possibly wrong) is that the current Matter SDK code does not automatically remove network credentials. This part is implemented by product makers in their applications.

Reply:
> there might be instances where the Thread device was commissioned to the Thread network out-of-band in the first place

Yes, so a per-spec implementation probably needs to keep track of how the credentials were configured.

> the current Matter SDK code does not automatically remove network credentials.

That's correct.